catalog/plugin/catalog-plugin-query-monitor-plugin/src/test/java/ddf/catalog/querymonitor/QueryMonitorTest.java

```diff
@@ -0,0 +1,259 @@
+
+            Iterator it = activeSearchTable.entrySet()
+                    .iterator();
+            HashMap.Entry<UUID, ActiveSearch> tempEntry = (HashMap.Entry) it.next();
+            ActiveSearch as = tempEntry.getValue();
+            UUID u = null;
+            Assert.assertFalse(qmpi.removeActiveSearch(u));
+            u = as.getUniqueID();
+            Assert.assertTrue(qmpi.getSearchTable()
+                    .containsKey(u));
+        } catch (Exception e) {
+            LOGGER.debug("Exception caught in testQueryMonitorPluginImplProcess in process method!");
+            Assert.fail();
+        }
+    }
+
+    @Test
+    public void testQueryMonitorPluginImplRemoveActiveSearchUsingUUIDValidUUID() {
+        QueryRequest mockQR = mock(QueryRequest.class);
+        propertyMap = new HashMap<>();
+        when(mockQR.getProperties()).thenReturn(propertyMap);
+        when(mockQR.toString()).thenReturn("garbageText");
+        qmpi.setRemoveFromTable(false);
+        try {
+            qmpi.process(null, mockQR);
+            HashMap<UUID, ActiveSearch> activeSearchTable = qmpi.getSearchTable();
+
+            Iterator it = activeSearchTable.entrySet()
+                    .iterator();
+            HashMap.Entry<UUID, ActiveSearch> tempEntry = (HashMap.Entry) it.next();
+            ActiveSearch as = tempEntry.getValue();
+            UUID u = as.getUniqueID();
+            Assert.assertTrue(qmpi.removeActiveSearch(u));
+        } catch (Exception e) {
+            LOGGER.debug("Exception caught in testQueryMonitorPluginImplProcess in process method!");
+            Assert.fail();
+        }
+    }
+
+    @Test
+    public void testQueryMonitorAddActiveSearchNullParameters() {
+        QueryRequest mockQR = mock(QueryRequest.class);
+        propertyMap = new HashMap<>();
+        when(mockQR.getProperties()).thenReturn(propertyMap);
+        when(mockQR.toString()).thenReturn("garbageText");
+        ActiveSearch as = new ActiveSearch(null, null, null);
+        qmpi.addActiveSearch(as);
+        HashMap<UUID, ActiveSearch> activeSearchTable = qmpi.getSearchTable();
+        Assert.assertTrue(activeSearchTable.containsValue(as));
+    }
+
+    @Test
+    public void testQueryMonitorPluginImplPreFedProcessNullQueryRequest() {
+        CatalogProvider mockSource = mock(CatalogProvider.class);
+        try {
+            //    qmpi.process(null, mockQR);
+            qmpi.process(mockSource, null);
+        } catch (Exception e) {
+            LOGGER.debug("Exception caught in testQueryMonitorPluginImplProcess in process method!");
+            Assert.fail();
+        }
+
+    }
+
+    @Test
+    public void testQueryMonitorPluginImplPreFedProcessNullSource() {
+        QueryRequest mockQR = mock(QueryRequest.class);
+        propertyMap = new HashMap<>();
+        when(mockQR.getProperties()).thenReturn(propertyMap);
+        when(mockQR.toString()).thenReturn("garbageText");
```

Comment:
Since "garbageText" and "someQuery" are used in multiple spots, it may be better to have it as a global static final constant